examples/workspaces_variables/manager/main.tf

```diff
@@ -0,0 +1,156 @@
+# This configuration creates and manages workspaces in Terraform Cloud /
+# Enterprise. Workspaces and the variables that should be set on them come from
+# two maps, and can come from additional sources as well, such as individual
+# variable resource blocks. The maps are merged, and then the result is
+# iterated over using for_each.
+
+variable "tf_organization" {
+  description = "The Terraform Cloud or Enterprise organization under which all operations should be performed."
+}
+
+variable "vcs_repo_identifier" {
+  description = <<-EOT
+  The format of VCS repo identifier might differ depending on the VCS provider,
+  see https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/workspace
+  EOT
+}
+
+variable "vcs_token" {
+  description = "The VCS token should correspond to an API token that can create OAuth clients."
+}
+
+variable "vars_mapped_by_workspace_name" {
+    description = <<-EOT
+    This is the map of workspaces and variables. A workspace is created for each
+    top level key and then variables are set on the workspace.
+    EOT
+}
+
+variable "additional_vars" {
```

Comment:
These variables should specify their type. Given this is an example for those starting to manage TFC workspaces we should eliminate as many assumptions as we can.